src/app/ui/editor/editor.cpp

```diff
@@ -1145,21 +1148,156 @@ void Editor::drawGrid(Graphics* g,
   grid_color =
     gfx::rgba(gfx::getr(grid_color), gfx::getg(grid_color), gfx::getb(grid_color), alpha);
 
-  // Draw horizontal lines
-  int x1 = spriteBounds.x;
-  int y1 = gridF.y;
-  int x2 = spriteBounds.x + spriteBounds.w;
-  int y2 = spriteBounds.y + spriteBounds.h;
+  // Orthogonal grid
+  if (isPixelGrid || getSite().sprite()->gridType() == doc::Grid::Type::Orthogonal) {
```

Comment:
We have to avoid calling `getSite()` as much as possible (the site collects a lot of information that might be unnecessary):
```suggestion
  if (isPixelGrid || m_sprite->gridType() == doc::Grid::Type::Orthogonal) {
```